app/views/homepage/_location_bar.haml

```diff
@@ -8,12 +8,17 @@
     function initialize() {
       var input = document.getElementById('q');
       var autocomplete = new google.maps.places.Autocomplete(input);
+      autocomplete.setTypes(['geocode']);
 
       google.maps.event.addListener(autocomplete, 'place_changed', function(event){
         var place = autocomplete.getPlace();
-        if(place != null && place.name != "") {
-          console.log('name:', place.name, '\naddress: ', place.formatted_address, '\ncoordinates: ', place.geometry.location.toUrlValue());
-          document.getElementById('lc').value = place.geometry.location.toUrlValue();
+        if(place != null) {
+          if( place.geometry != null ) {
+            console.log('name:', place.name, '\naddress: ', place.formatted_address, '\ncoordinates: ', place.geometry.location.toUrlValue());
+            document.getElementById('lc').value = place.geometry.location.toUrlValue();
+          } else {
+            console.log("Please select some place from the autocomplete list.")
+          }
         }
         document.getElementById('homepage-filters').submit();
```

Comment:
This is already a good amount of JavaScript. Should it be in it's own .js file instead of inlining?